### ISSUE TITLE
RecursionError when running tests

### ISSUE THREAD

**Issue by hadalin** [avinassh/haxor#21]
Running tests:
```
pip install -r requirements-dev.txt
python setup.py develop
pytest tests
```

produces the following error
```
RecursionError: maximum recursion depth exceeded while calling a Python object
```

Python version: `3.6.5`

**Comment by avinassh** [comment]
Are you still getting this error? Builds are passing on Travis CI - https://travis-ci.org/avinassh/haxor

**Comment by hadalin** [comment]
Yes.

Steps to reproduce:
- `mkvirtualenv haxor -p ~/.pyenv/versions/3.6.5/bin/python`
- `pip install -r requirements-dev.txt` (raises `pytest 3.5.1 has requirement py>=1.5.0, but you'll have py 1.4.29 which is incompatible.`)
- `python setup.py develop`
- `pytest tests`

**Comment by avinassh** [comment]
fixed from #32
